.github/workflows/schedule-update-actions.yml

```diff
@@ -0,0 +1,26 @@
+# SPDX-FileCopyrightText: 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+
+name: GitHub Actions Version Updater
+on:
+  workflow_dispatch:
+  schedule:
+    # Automatically run on every Sunday
+    - cron:  '0 0 * * 0'
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - uses: actions/checkout@v3.5.2
+        with:
+          # [Required] Access token with `workflow` scope.
+          token: ${{ secrets.PAT }}
+
+      - name: Run GitHub Actions Version Updater
+        uses: saadmk11/github-actions-version-updater@v0.8.1
+        with:
+          # [Required] Access token with `workflow` scope.
+          token: ${{ secrets.PAT }}
+          pull_request_title: "ci: Update GitHub Actions to Latest Version"
```

Comment:
If we would enable `skip_pull_request` to not push PRs for updates, would it still also require the token?

I'd suggest using `update_version_with: 'release-commit-sha'` as stated in https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions and also https://github.com/ossf/scorecard/blob/main/docs/checks.md#pinned-dependencies

Reply:
Yes!! would still likely require a token  .. :bowtie: